br/pkg/storage/gcs.go

```diff
@@ -319,12 +321,26 @@ func NewGCSStorage(ctx context.Context, gcs *backuppb.GCS, opts *ExternalStorage
 	if gcs.Endpoint != "" {
 		clientOps = append(clientOps, option.WithEndpoint(gcs.Endpoint))
 	}
-	// the HTTPClient should has credential, currently the HTTPClient only has the http.Transport.
-	// So we remove the HTTPClient in the storage.New().
-	// Issue: https: //github.com/pingcap/tidb/issues/47022
-	if opts.HTTPClient != nil {
-		clientOps = append(clientOps, option.WithHTTPClient(opts.HTTPClient))
+
+	httpClient := opts.HTTPClient
+	if httpClient == nil {
+		// http2 will to reuse the connection to read multiple files, which is
+		// very slow, the speed is about the same speed as reading a single file.
+		// So we disable keepalive here to use multiple connections to read files.
```

Comment:
not sure if it should be the default behaviour of Google clous storage  
cc @Leavrth @YuJuncen

Reply:
I think it's OK because we don't have a high concurrency of Google cloud Storage objects accessing, and the socket/memory usage should be affordable.